homeassistant/components/huum/climate.py

```
@@ -0,0 +1,131 @@
+            return "mdi:radiator"
+        return "mdi:radiator-off"
+
+    @property
+    def current_temperature(self) -> float | None:
+        """Return the current temperature."""
+        return self._status.temperature
+
+    @property
+    def target_temperature(self) -> int:
+        """Return the temperature we try to reach."""
+
+        return self._status.target_temperature or self._target_temperature
+
+    async def async_set_hvac_mode(self, hvac_mode: str) -> None:
+        """Set hvac mode."""
+        if hvac_mode == HVACMode.HEAT:
+            temperature = max(self.min_temp, self.target_temperature)
+            await self._turn_on(temperature)
+        elif hvac_mode == HVACMode.OFF:
+            await self._huum_handler.turn_off()
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set new target temperature."""
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+        if temperature is None:
+            return
+        self._target_temperature = temperature
+
+        if self.hvac_mode == HVACMode.HEAT:
+            await self._turn_on(temperature)
+
+    async def async_update(self) -> None:
+        """Get the latest status data.
+
+        We get the latest status first from the status endpoints of the sauna.
+        If that data does not include the temperature, that means that the sauna
+        is off, we then call the off command which will in turn return the temperature.
+        This is a workaround for getting the temperature as the Huum API does not
+        return the target temperature of a sauna that is off, even if it can have
+        a target temperature at that time.
+        """
+        self._status = await self._huum_handler.status_from_status_or_stop()
+
+    async def _turn_on(self, temperature) -> None:
+        try:
+            await self._huum_handler.turn_on(temperature)
+        except (ValueError, SafetyException) as turn_on_error:
+            _LOGGER.error(str(turn_on_error))
```

I think you want to throw `HomeAssistantError` rather than swallowing the exception e.g. so an automation would fail